static/panes/compiler.js

```diff
@@ -1171,6 +1168,11 @@ Compiler.prototype.setAssembly = function (result, filteredCount) {
     this.assembly = asm;
     if (!this.outputEditor || !this.outputEditor.getModel()) return;
     var editorModel = this.outputEditor.getModel();
+    if (result.languageId) {
+        monaco.editor.setModelLanguage(editorModel, result.languageId);
+    } else {
+        monaco.editor.setModelLanguage(editorModel, 'asm');
```

Comment:
Shouldn't this be the removed `(languages[this.currentLangId] ? languages[this.currentLangId].monacoDisassembly : null) || 'asm'`? 

As it stands right now, languages that rely on monacoDissasembly to correclty work (C++ implements its own `cppp` mode to fix some issues in Monaco's, for example) won't work

Reply:
I have added the `getCompilerResultLanguageId` method in `jakt.ts` and `cppfront.ts`. It should fix the issue you are mentioning here.